source/Mlos.Python/mlos/OptimizerEvaluationTools/OptimizerEvaluator.py

```diff
@@ -0,0 +1,218 @@
+from mlos.Optimizers.OptimumDefinition import OptimumDefinition
+from mlos.Optimizers.RegressionModels.GoodnessOfFitMetrics import DataSetType
+from mlos.Optimizers.RegressionModels.RegressionModelFitState import RegressionModelFitState
+from mlos.Spaces import Point
+from mlos.Tracer import trace, traced, Tracer
+
+
+class OptimizerEvaluator:
+    """Evaluates optimizers against objective functions.
+
+    This class are responsible for:
+        1. instantiating an optimizer
+        2. instantiating an objective function
+        3. launching an optimizer against that objective function
+        4. keeping track of the goodness of fit, and of the various optima over time
+        5. producing an instance of OptimizerEvaluationReport.
+
+    """
+
+    def __init__(
+        self,
+        optimizer_evaluator_config: Point,
+        optimizer: OptimizerBase = None,
+        optimizer_config: Point = None,
+        objective_function: ObjectiveFunctionBase = None,
+        objective_function_config: Point = None,
+    ):
+        assert optimizer_evaluator_config in optimizer_evaluator_config_store.parameter_space
+        assert (optimizer is None) != (optimizer_config is None), "A valid optimizer XOR a valid optimizer_config must be supplied."
+        assert (objective_function is None) != (objective_function_config is None),\
+            "A valid objective_function XOR a valid objective_function_config must be specified"
+
+        self.optimizer_evaluator_config = optimizer_evaluator_config
+        self.objective_function_config = None
+        self.objective_function = None
+        self.optimizer_config = None
+        self.optimizer = None
+
+        # Let's get the objective function assigned to self's fields.
+        #
+        if (objective_function_config is not None) and (objective_function is None):
+            assert objective_function_config in objective_function_config_store.parameter_space
+            self.objective_function_config = objective_function_config
+            self.objective_function = ObjectiveFunctionFactory.create_objective_function(objective_function_config)
+
+        elif (objective_function is not None) and (objective_function_config is None):
+            self.objective_function_config = objective_function.objective_function_config
+            self.objective_function = objective_function
+
+        else:
+            # The assert above should have caught it but just in case someone removes or changes it.
+            #
+            assert False, "A valid objective_function XOR a valid objective_function_config must be specified"
+
+        # Let's get the optimzier and its config assigned to self's fields.
+        #
+        if (optimizer_config is not None) and (optimizer is None):
+            assert optimizer_config in bayesian_optimizer_config_store.parameter_space
+
+            objective_name = self.objective_function.output_space.dimension_names[0]

```

Comment:
Add a work item to track this - we will need to change it for multi-objective optimization.

Reply:
#159